mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -281,9 +265,6 @@ class GeneralMLWriter(stage: PipelineStage) extends MLWriter with Logging {
 
   // override for Java compatibility
   override def session(sparkSession: SparkSession): this.type = super.session(sparkSession)
-
-  // override for Java compatibility
-  override def context(sqlContext: SQLContext): this.type = super.session(sqlContext.sparkSession)
```

Comment:
@srowen This public method seems had not been deprecated before removal, and is avaiable in 2.4.5.
```
scala> import org.apache.spark.ml.util.GeneralMLWriter
import org.apache.spark.ml.util.GeneralMLWriter

scala> new GeneralMLWriter(null).context(spark.sqlContext)
res0: org.apache.spark.ml.util.GeneralMLWriter = org.apache.spark.ml.util.GeneralMLWriter@26b150cd
```
There is no deprecation warning above. Does it matter?

Reply:
This seems properly deprecated in `MLWriter` as its parent.
![Screen Shot 2020-02-26 at 3 57 54 PM](https://user-images.githubusercontent.com/6477701/75319873-d0df3980-58b0-11ea-9fbc-7cbb9510acc1.png)

and the Scaladoc explicitly shows this too. Seems right to remove together if we should in `MLWriter`.

Reply:
Yeah it was deprecated in 2.0.0 and marked for removal in 3.0.0.

```
 /**	
   * Sets the Spark SQLContext to use for saving/loading.	
   *	
   * @deprecated Use session instead. This method will be removed in 3.0.0.	
   */	
  @Since("1.6.0")	
  @deprecated("Use session instead. This method will be removed in 3.0.0.", "2.0.0")
```

I think ideally the Java overload and subclass overrides would be marked deprecated too, but they implicitly are. If there were a case that this is actually used, we could revive it, but just wondering how often people would be using save + SQLContext?

Reply:
I never use this method, just check it. Thanks!